intrastat_product/models/hs_code.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2011-2020 Akretion (http://www.akretion.com)
+# Copyright 2009-2020 Noviat (http://www.noviat.com)
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# @author Luc de Meyer <info@noviat.com>
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HSCode(models.Model):
+    _inherit = "hs.code"
+
+    intrastat_unit_id = fields.Many2one(
+        comodel_name="intrastat.unit", string="Intrastat Supplementary Unit"
+    )
+
+    @api.constrains("local_code")
+    def _hs_code(self):
+        eu_countries = self.env.ref("base.europe").country_ids
```

Comment:
I don't think it's a good idea to base a constraint on base.europe since the data in that country group is not guaranteed to be correct (noupdate=1).
I suppose the reason behind this is that some non-eu countries extend the hs_code with 'non-digits.

I propose we simply drop the isdigits check here and move the check to the generation of the declaration.

Reply:
Is Intrastat used outside Europe. If not, why make the check only in this case? Let's put the constraint applied universally.

Reply:
@pedrobaeza 
Intrastat is within europe (+nothern ireland) but hs_code is international hence we shouldn't add constraints breaking use of hs_code outside europe. That why I propose to remove the constraint here and add it as a check when generating the declaration.

Reply:
OK, I see. But reading https://en.wikipedia.org/wiki/Harmonized_System, it seems this classification is normalized across the world, and there aren't specific Europe constraints, isn't it?

Reply:
Another thing that I'm thinking of is that `product_harmonized_system` should be in OCA/product-attribute or OCA/community-data-files then.

Reply:
@pedrobaeza 
1) european constraints: the european countries extend the number with digits. I am not sure if the rest of the world extend the codes with digits or characters. @rvalyi : do you have a better view on this ?

2) product-attribute
I am not sure. This repo is called intrastat-extrastat since it goes beyond europe.
For a recent project with import for Asia we have done some work to extend the HS Codes with a country table and add the country specific duties. This work in intended to end up in this repo as well.

Reply:
OK, then just let's remove them for now and we decide later. At the end, for most of us, the codes are imported via a wizard, so it's not a great problem. cc @joao-p-marques